src/group_theory/free_group.lean

```diff
@@ -863,4 +884,64 @@ fintype.subtype (list.to_finset $ red.enum L₁) $
 
 end reduce
 
+section metric
+
+variable [decidable_eq α]
+
+/-- The length of reduced words provides a norm on a free group. --/
+def norm (x : free_group α) : nat := x.to_word.length
+
+lemma norm_inv_le (x : free_group α) : norm x⁻¹ ≤ norm x :=
+begin
+  let w := x.to_word,
+ 
+  calc norm x⁻¹ = norm ((mk (x.to_word))⁻¹) : by simp only [to_word.mk]
+       ... = norm (mk (inv_rev w)) : by simp only [←inv_mk]
+       ... ≤ (inv_rev w).length : by exact to_word.le_length
+       ... = w.length : inv_rev_length
+       ... = norm x : rfl
+end
+
+lemma norm_inv_eq (x : free_group α) : norm x = norm x⁻¹ :=
```

Comment:
If you swap LHS with RHS, then this will be a good `@[simp]` lemma.

Reply:
I've swapped this and made this into a @[simp] lemma in 2aa2cdf7b0b6d5c07dd6e854897c155e60dc975a.